### PR TITLE
87736992 approve reject cart show

### PR DESCRIPTION
**Pull request by phirefly**
WIP: Feedback would be great, especially around bypassing token validation through the use of a queryparam. I'm starting to feel the pain of managing role, so that work will likely clean up the checks for cart status in the carts helper.

### CODE REVIEW

File: app/controllers/communicarts_controller.rb [hunk 1]
```diff
@@ -52,6 +55,9 @@ def approval_response
 private
 
   def validate_access
+    request_from_email = params['email_delivery'].presence || true
+    return unless request_from_email == true
```
Comment (cmc333333) [inline app/controllers/communicarts_controller.rb]: This logic is kind of confusing. If `params['email_delivery']` is not truthy, we'll get `true` (due of the second half of the or); we'll then return. If `params['email_delivery']` *is* truthy, we won't return. It doesn't look like it's ever truthy, though?

File: spec/steps/approval_steps.rb [hunk 2]
```diff
@@ -12,13 +16,24 @@ module ApprovalSteps
...
+    @cart.approvals << FactoryGirl.create(:approval, role: 'approver', user_id: approver.id, position: position)
   end
 
   step "feature flag :flag_name is :value" do |flag, value|
     ENV[flag] = value
   end
 
+  step 'the cart has been approved by the logged in user' do
+    approval = @cart.approvals.where(user_id: @current_user.id).first
+    approval.update_attributes(status: 'approved')
```
Comment (cmc333333) [inline spec/steps/approval_steps.rb]: Consider `approval.approve!` for consistency.

File: app/controllers/communicarts_controller.rb (hunk 1 above)
Comment (phirefly) [inline app/controllers/communicarts_controller.rb]: Yeah, it is confusing. From the web requests, I'm using a queryparam `email_delivery=false` which bypasses the token validation.

Comment (phirefly) [inline app/controllers/communicarts_controller.rb]: I can probably simplify to `return if params['email_delivery'] == false`

Comment (phirefly) [inline app/controllers/communicarts_controller.rb]: So, to answer your previous question, the current code sets `request_from_email` to true unless we explicitly say it's false, which is what I'm doing in the request. If we don't pass the param, the default value is true. This is the case when we click Approve from one of the emails.

File: spec/steps/approval_steps.rb (hunk 2 above)
Comment (phirefly) [inline spec/steps/approval_steps.rb]: I don't see that method. But it should exist!

Comment (phirefly) [inline spec/steps/approval_steps.rb]: Perhaps it's avilable after the merge? ;)

Comment (cmc333333) [inline spec/steps/approval_steps.rb]: Yep; after you merge master, you'll have access to the workflow methods, including `approve!`

File: app/controllers/communicarts_controller.rb (hunk 1 above)
Comment (cmc333333) [inline app/controllers/communicarts_controller.rb]: `return if params['email_delivery'] == false` is more readable to my untrained eye.

Bigger question: why is whether or not a user came from email the deciding factor? I'd expect the token to be ignored (though, perhaps still expired) if the user was already logged in, regardless of how the user reached this page.

**Comment by phirefly** [comment]
Squashing and submitting another PR

**Pull request closed without merging**